### PR TITLE
make it easier to use the odb index for traversals

### PR DESCRIPTION
**Pull request by bbrehm**
This is related to https://github.com/ShiftLeftSecurity/overflowdb/pull/309

Version will be fixed once upstream is merged.

### CODE REVIEW

File: semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/NodeTypeStarters.scala [hunk 1]
```diff
@@ -145,25 +145,25 @@ class NodeTypeStarters(cpg: Cpg) extends TraversalSource(cpg.graph) {
...
-    cpg.graph.nodes(NodeTypes.METHOD).cast[Method]
+    InitialTraversal.from[Method](cpg.graph, NodeTypes.METHOD)
 
-  /** Shorthand for `cpg.method.name(fullName)`
+  /** Shorthand for `cpg.method.name(name)`
     */
-  @Doc(info = "All methods with given name")
-  def method(name: String): Traversal[Method] =
-    method.name(name)
+  @Doc(info = "All methods with name matchine the given pattern")
```
Comment (mpollmeier) [inline semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/NodeTypeStarters.scala:152]:
```suggestion
  @Doc(info = "All methods with a name that matches the given pattern")
```

Comment (mpollmeier) [reply semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/NodeTypeStarters.scala:152]: (fixing the typo, but also minor improvement of readabity imo)

File: joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/TypeTests.scala [hunk 2]
```diff
@@ -155,6 +155,9 @@ class TypeTests extends JavaSrcCodeToCpgFixture {
   }
 
   "should default to ANY with a matching type node for unresolved types" in {
+    val xxx = cpg.typ.fullName("ANY").l
+    val yyy = cpg.typ.filter(_ => true).fullName("ANY").l
+    val zzz = 1 + 1
```
Comment (mpollmeier) [inline joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/TypeTests.scala:160]: what's this?

**Review by mpollmeier — COMMENTED** [review]
we have another `NodeTypeStarters` class: https://github.com/joernio/joern/blob/master/semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/operatorextension/NodeTypeStarters.scala

File: joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/TypeTests.scala (hunk 2 above)
Comment (bbrehm) [reply joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/TypeTests.scala]: It's debug nonsense, thanks for spotting, removed.
(intellij insists on some lines with instructions in order to set breakpoints)

**Comment by bbrehm** [comment]
>we have another NodeTypeStarters class:

I don't think there is anything to change there. Furthermore, the OperatorExtensions are a super dirty legacy hack that I won't touch if at all avoidable. They fundamentally lie to the graph by wrapping a second NodeRef around the same NodeDb.

**Comment by fabsx00** [comment]
The operator extension is an example of what's called a decorator language, and it isn't a hack nor is it legacy. There is no "lying" involved either.

**Review by mpollmeier — APPROVED** [review]

**Pull request merged**